vita3k/modules/SceCommonDialog/SceCommonDialog.cpp

```diff
@@ -805,54 +805,52 @@ EXPORT(int, sceSaveDataDialogAbort) {
             emuenv.common_dialog.savedata.icon_buffer[idx] = thumbnail_buffer;
-            emuenv.common_dialog.savedata.icon_loaded[idx] = true;
         } else if (iconBuf && iconBufSize != 0) {
             thumbnail_buffer.insert(thumbnail_buffer.end(), iconBuf, iconBuf + iconBufSize);
             emuenv.common_dialog.savedata.icon_buffer[idx] = thumbnail_buffer;
-            emuenv.common_dialog.savedata.icon_loaded[idx] = true;
-        } else {
-            emuenv.common_dialog.savedata.icon_loaded[idx] = false;
         }
-    } else {
+    } else
         emuenv.common_dialog.savedata.title[idx] = emuenv.common_dialog.lang.save_data.save["new_saved_data"];
-        emuenv.common_dialog.savedata.icon_loaded[idx] = false;
-    }
 }
 
-static void check_save_file(SceUID fd, std::vector<SceAppUtilSaveDataSlotParam> slot_param, int index, EmuEnvState &emuenv, const char *export_name) {
-    vfs::FileBuffer thumbnail_buffer;
+static void check_save_file(const uint32_t index, EmuEnvState &emuenv, const char *export_name) {
+    SceUID fd = open_file(emuenv.io, construct_slotparam_path(emuenv.common_dialog.savedata.slot_id[index]).c_str(), SCE_O_RDONLY, emuenv.pref_path, export_name);
     if (fd < 0) {
         auto empty_param = emuenv.common_dialog.savedata.list_empty_param[index];
         check_empty_param(emuenv, empty_param, index);
     } else {
-        read_file(&slot_param[index], emuenv.io, fd, sizeof(SceAppUtilSaveDataSlotParam), export_name);
+        vfs::FileBuffer thumbnail_buffer;
+        SceAppUtilSaveDataSlotParam slot_param;
+        read_file(&slot_param, emuenv.io, fd, sizeof(SceAppUtilSaveDataSlotParam), export_name);
         close_file(emuenv.io, fd, export_name);
         emuenv.common_dialog.savedata.slot_info[index].isExist = 1;
-        emuenv.common_dialog.savedata.title[index] = slot_param[index].title;
-        emuenv.common_dialog.savedata.subtitle[index] = slot_param[index].subTitle;
-        emuenv.common_dialog.savedata.details[index] = slot_param[index].detail;
-        emuenv.common_dialog.savedata.date[index] = slot_param[index].modifiedTime;
+        emuenv.common_dialog.savedata.title[index] = slot_param.title;
+        emuenv.common_dialog.savedata.subtitle[index] = slot_param.subTitle;
+        emuenv.common_dialog.savedata.details[index] = slot_param.detail;
+        emuenv.common_dialog.savedata.date[index] = slot_param.modifiedTime;
         emuenv.common_dialog.savedata.has_date[index] = true;
-        auto device = device::get_device(slot_param[index].iconPath);
-        auto thumbnail_path = translate_path(slot_param[index].iconPath, device, emuenv.io.device_paths);
+        auto device = device::get_device(slot_param.iconPath);
+        auto thumbnail_path = translate_path(slot_param.iconPath, device, emuenv.io.device_paths);
         vfs::read_file(VitaIoDevice::ux0, thumbnail_buffer, emuenv.pref_path, thumbnail_path);
```

Comment:
We got device and read from ux0. It looks a bit strange.

Reply:
not my code here haha